runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -2125,14 +2155,25 @@ void init_ofiEp(void) {
     avAttr.count *= numTxCtxs;
   }
 
-  OFI_CHK(fi_av_open(ofi_domain, &avAttr, &ofi_av, NULL));
+  // The EFA provider does not allow AVs to be shared among
+  // endpoints. For all other providers we'll share them to reduce
+  // overhead.
+
+  chpl_bool useSharedAv = !providerInUse(provType_efa);
+  if (useSharedAv) {
```

Comment:
Further down at (new) line 2275 the control flow is opposite: an outer if-stmt based on `useScalEp` surrounding an inner if-stmt based on `ofi_av == NULL`. That made this a little hard to follow for me, and I'm still not totally convinced it's right, especially if the cxi provider someday supported scalable endpoints. Maybe just take the hit on code redundancy, and put whatever control flow we need for AVs into both branches of the `if (useScalEp)` below, instead of doing it on its own here?